src/main/java/com/reservation/payment/domain/exception/CurrencyMismatchException.java

```diff
@@ -0,0 +1,14 @@
+package com.reservation.payment.domain.exception;
+
+import java.util.Currency;
+
+public class CurrencyMismatchException extends RuntimeException {
+
+    public CurrencyMismatchException(Currency expected, Currency actual) {
+        super(message(expected, actual));
+    }
+
+    public static String message(Currency expected, Currency actual) {
```

Comment:
```suggestion
    private static String message(Currency expected, Currency actual) {
```

Ta metodka chyba nie jest nigdzie używana poza tą klasą?

A w ogóle to widzę, że w wyjątkach raczej używamy stałej `MESSAGE`, więc czemu tutaj akurat zrobiłeś to jako metodę?
Dla mnie obie opcje spoko, ale zdecydujmy się na jedną 😁